### DISCUSSION TITLE
Flutter Mix

### DISCUSSION THREAD

**Discussion by codscino** [nank1ro/flutter-shadcn-ui#32] [Ideas]
Just read this from Code with Andrea newsletter, I am not an expert but maybe this could be useful:

> While web developers have enjoyed custom UI kits such as RadizUi and shacdn, Flutter devs have been limited to the Material and Cupertino widgets for some time.

>Enter Flutter Mix: a new package offering expressive building blocks for building design systems in Flutter.
When compared to the usual Flutter styling code, Mix offers a more streamlined syntax, and while it is still in beta, it seems promising.

**Comment by nank1ro** [comment]
I've seen Mix's documentation, it's a great idea for writing UI. It allows you to write code in a similar way to SwiftUI and very far from Flutter's style.
Is a very interesting approach, but for the progress at which the library is, it will certainly continue on the Flutter-style path.
It would be a lot of work to rewrite the components already implemented, and bugs would probably appear due to the mix library.
It will be curious to see if Flutter Mix will mature and if it will really be used by the community.

Let's see how it goes, and if interest will increase 👍🏼

**Comment by codscino** [comment]
I completely agree. Lets see

**Comment by pythonhubdev** [comment]
Hi @nank1ro I recently came accross Flutter Mix and I wanted to use it for creating some custom components like kind of shared styling or utilities which I can re-use but I have already developed my app with `flutter_shadcn` so I thought of creating some variants of `flutter_shadcn` with `mix` but that wasn't possible because `mix` has it's own widgets. 

If you take the official Shadcn it's written in combination with TailwindCSS but we can also apply generic CSS to it as well by passing in class name so I thought it would be good to have `flutter_shadcn` built on top of `mix` very similar to react_shadcn + tailwindcss. But if there is a way we can allow people to apply normal styling as well. This could be complicated or hard to implement but putting my thoughts in. May be some kind of extension  should work.

**Comment by medz** [comment]
https://github.com/nank1ro/flutter-shadcn-ui/discussions/539#discussioncomment-14948537

If we do that, then it will become possible!

**Reply by nank1ro** [reply]
Not exactly, Mix requires you to use some specific widgets, like Pressable, StyleBuilder and so on.
It would work only if we choose to use Mix
